dev/ScrollViewer/ScrollViewer.cpp

```diff
@@ -825,8 +882,30 @@ void ScrollViewer::OnHideIndicatorsTimerTick(
 {
     SCROLLVIEWER_TRACE_VERBOSE(*this, TRACE_MSG_METH, METH_NAME, this);
 
-    StopHideIndicatorsTimer(false /*isForDestructor*/);
-    HideIndicators(true /*useTransitions*/);
+    ResetHideIndicatorsTimer();
+
+    if (AreScrollControllersAutoHiding())
+    {
+        HideIndicators();
+    }
+}
+
+void ScrollViewer::OnAutoHideScrollBarsChanged(
+    winrt::UISettings const& uiSettings,
+    winrt::UISettingsAutoHideScrollBarsChangedEventArgs const& args)
+{
+    MUX_ASSERT(SharedHelpers::Is19H1OrHigher());
+
+    // OnAutoHideScrollBarsChanged is called on a non-UI thread, process notification on the UI thread using a dispatcher.
+    m_dispatcherHelper.RunAsync([strongThis = get_strong()]()
```

Comment:
Don't you want a weak ref that you check on execute? Doesn't seem like the async should hold the object alive.

Reply:
This looks correct to me. The RatingControl and ColorPicker controls are doing this as well.